### PR TITLE
Add docs for Grove Multichannel Gas Sensor V2

### PR DESCRIPTION
**Pull request by YorkshireIoT**
## Description:

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#4304

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/grove_gas_mc_v2.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    Grove Multichannel Gas Sensor V2
+
+The communication with this sensor is done via :ref:`I²C Bus <i2c>`, so you need to have
+an ``i2c:`` section in your config for this integration to work.
+
+.. code-block:: yaml
+
+    i2c:
+      sda: D2
+      scl: D1
```
Comment (jesserockz) [inline components/sensor/grove_gas_mc_v2.rst]: No board specific pins here please. Use `GPIO` numbers or remove the i2c section as it is already stated above as a requirement.

File: components/sensor/grove_gas_mc_v2.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        update_interval: 60s
+
+Configuration variables:
+------------------------
+
+- **no2** (**Required**): The Nitrogen Dioxide sensor data.
+
+  - **name** (**Required**, string): The name for the Nitrogen Dioxide sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
```
Comment (jesserockz) [inline components/sensor/grove_gas_mc_v2.rst]: No need to specify what is already in the sensor config section linked. This way "forces" people to learn the sensor core config better =)

Same for the rest please.

```suggestion
- **no2** (**Required**): The Nitrogen Dioxide sensor data. All options from :ref:`Sensor <config-sensor>`.
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Comment by YorkshireIoT** [comment]
> There hasn't been any activity on this pull request recently. This pull request has been automatically marked as stale because of that and will be closed if no further activity occurs within 7 days. Thank you for your contributions.

Waiting on review

**Comment by YorkshireIoT** [comment]
Ping

**Comment by YorkshireIoT** [comment]
Not stale

**Comment by YorkshireIoT** [comment]
Not stale

**Comment by YorkshireIoT** [comment]
Not stale

File: components/sensor/grove_gas_mc_v2.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        no2:
+          name: "Nitrogen Dioxide"
+        ethanol:
+          name: "Ethanol"
+        carbon_monoxide:
+          name: "Carbon Monoxide"
+        tvoc:
+          name: "Volatile Organic Compounds"
+        address: 0x08
+        update_interval: 60s
```
Comment (jesserockz) [inline components/sensor/grove_gas_mc_v2.rst]: Not needed
```suggestion
```

**Review by jesserockz — APPROVED** [review]

**Pull request merged**